frontend/build.go

```diff
@@ -13,13 +14,28 @@ import (
 	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+const (
+	windowsTargetKey    = "windowscross"
+	windowsTargetPrefix = windowsTargetKey + "/"
+)
+
 func loadSpec(ctx context.Context, client *dockerui.Client) (*dalec.Spec, error) {
 	src, err := client.ReadEntrypoint(ctx, "Dockerfile")
 	if err != nil {
 		return nil, fmt.Errorf("could not read spec file: %w", err)
 	}
 
-	spec, err := dalec.LoadSpec(bytes.TrimSpace(src.Data))
+	opts := []dalec.LoadOpt{}
+	if strings.HasPrefix(client.Target, windowsTargetPrefix) {
```

Comment:
Currently validation happens at load time. I will move this into the handler, it makes sense that each handler can do its own additional validations after the main validation is done.

Reply:
I think that is a good idea. From a delegation standpoint, it seems like `loadSpec` should check for validity at the Dalec spec level, and if individual handlers want to do further validation then they can

Reply:
Done.

I added validations to the `targetWrapper` so that validations functions can be registered alongside handlers. These are automatically run before the call to `Build`.